sklearn/cluster/_kmeans.py

```diff
@@ -225,47 +232,49 @@ def k_means(X, n_clusters, sample_weight=None, init='k-means++',
 
         False : never precompute distances
 
-    n_init : int, optional, default: 10
+        .. deprecated:: 0.23
+            'precompute_distances' was deprecated in version 0.23 and will be
+            removed in 0.25.
+
+    n_init : int, (default=10)
         Number of time the k-means algorithm will be run with different
         centroid seeds. The final results will be the best output of
         n_init consecutive runs in terms of inertia.
 
-    max_iter : int, optional, default 300
+    max_iter : int, (default=300)
         Maximum number of iterations of the k-means algorithm to run.
 
-    verbose : boolean, optional
+    verbose : boolean, optional (default=False)
         Verbosity mode.
 
-    tol : float, optional
+    tol : float (default=1e-4)
         The relative increment in the results before declaring convergence.
 
-    random_state : int, RandomState instance or None (default)
+    random_state : int, RandomState instance or None (default=None)
         Determines random number generation for centroid initialization. Use
         an int to make the randomness deterministic.
         See :term:`Glossary <random_state>`.
 
-    copy_x : bool, optional
+    copy_x : boolean, optional (default=True)
         When pre-computing distances it is more numerically accurate to center
-        the data first.  If copy_x is True (default), then the original data is
-        not modified, ensuring X is C-contiguous.  If False, the original data
-        is modified, and put back before the function returns, but small
-        numerical differences may be introduced by subtracting and then adding
-        the data mean, in this case it will also not ensure that data is
-        C-contiguous which may cause a significant slowdown.
+        the data first. If copy_x is True (default), then the original data is
+        not modified. If False, the original data is modified, and put back
+        before the function returns, but small numerical differences may be
+        introduced by subtracting and then adding the data mean. Note that if
+        the original data is not C-contiguous, a copy will be made even if
+        copy_x is False. If the original data is sparse, but not in CSR format,
+        a copy will be made even if copy_x is False.
 
     n_jobs : int or None, optional (default=None)
-        The number of jobs to use for the computation. This works by computing
-        each of the n_init runs in parallel.
+        The number of jobs to use for the computation.
```

Comment:
This is not a number of jobs (the name is bad) but a number of worker threads. I think it's important to be explicit about the nature of the parallelism used for this algorithm: joblib is not used but instead one uses the OpenMP thread-based mechanism to parallelize the main Cython loop that assigns each sample to each closest cluster center.

Reply:
Should we just deprecate this parameter as discussed in #14265 ?

Reply:
For now I would keep it to avoid removing an existing feature. But we should be more explicit in the docstring on the nature of the parallelism involved here.